File: app/src/main/java/com/hjc/reader/ui/wan/adapter/TagAdapter.java
package com.hjc.reader.ui.wan.adapter;


import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.hjc.reader.R;
import com.hjc.reader.bean.response.WanListBean;
import com.hjc.reader.databinding.ItemTagBinding;


public class TagAdapter extends BaseQuickAdapter<WanListBean.DataBean.DatasBean, BaseViewHolder> {
    public TagAdapter() {
        super(R.layout.item_tag);
    }

    @Override
    protected void onItemViewHolderCreated(@NonNull BaseViewHolder viewHolder, int viewType) {
        DataBindingUtil.bind(viewHolder.itemView);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, WanListBean.DataBean.DatasBean item) {
        if (item == null) {
            return;
        }

        ItemTagBinding binding = helper.getBinding();
        if (binding != null) {
            binding.setArticleBean(item);
        }
    }
}

File: module-wan/src/main/java/com/hjc/learn/wan/adapter/TagAdapter.kt
package com.hjc.learn.wan.adapter

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.hjc.learn.wan.R
import com.hjc.learn.wan.databinding.WanItemTagBinding
import com.hjc.library_net.model.WanArticleBean

/**
 * @Author: HJC
 * @Date: 2021/2/22 17:14
 * @Description: 标签列表adapter
 */
class TagAdapter(data: MutableList<WanArticleBean>?) :
    BaseQuickAdapter<WanArticleBean, BaseViewHolder>(R.layout.wan_item_tag, data) {

    override fun onItemViewHolderCreated(viewHolder: BaseViewHolder, viewType: Int) {
        DataBindingUtil.bind<ViewDataBinding>(viewHolder.itemView)
    }

    override fun convert(holder: BaseViewHolder, item: WanArticleBean) {
        val binding = DataBindingUtil.getBinding<WanItemTagBinding>(holder.itemView)
        binding?.articleBean = item
    }
}